app/src/main/java/com/vultisig/wallet/ui/models/swap/SwapFormViewModel.kt

```diff
@@ -542,7 +543,25 @@ internal class SwapFormViewModel @Inject constructor(
             selectedSrc.value = selectedDst.value
             selectedDst.value = buffer
         }
+    }
+
+    fun selectSrcPercentage(percentage: Float) {
+        val selectedSrcAccount = selectedSrc.value?.account ?: return
+        val srcTokenValue = selectedSrcAccount.tokenValue ?: return
+
+        val srcToken = selectedSrcAccount.token
+
+        val swapFee =
+            quote?.fees?.value.takeIf { provider == SwapProvider.LIFI } ?: BigInteger.ZERO
+
+        val maxUsableTokenAmount =
+            srcTokenValue.value - swapFee - (gasFee.value?.value?.takeIf { srcToken.isNativeToken }
+                ?: BigInteger.ZERO)
+
+        val amount = TokenValue.createDecimal(maxUsableTokenAmount, srcTokenValue.decimals)
```

Comment:
can you please round up the number?  like when I click 50%, it get a huge number
![CleanShot 2025-05-01 at 19 00 18@2x](https://github.com/user-attachments/assets/b5fcf3ee-e452-4dcd-bfc9-01e794976e16)